snap/frontend/components/TransactionTable/TransactionTable.spec.jsx

```diff
@@ -0,0 +1,251 @@
+
+			// Act:
+			const icon = screen.getByAltText(expectedIcon);
+
+			// Assert:
+			expect(icon).toBeInTheDocument();
+		};
+
+		it('renders send icon when sender is the same as selected account', () => {
+			assertIcon(
+				{
+					sender: senderAddress
+				},
+				'Send logo'
+			);
+		});
+
+		it('renders receive icon when sender is not the same as selected account', () => {
+			assertIcon(
+				{
+					sender: recipientAddress
+				},
+				'Receive logo'
+			);
+		});
+	});
+
+	describe('second column', () => {
+		const assertTransactionType = (transaction, expectedTransactionType) => {
+			// Arrange:
+			context.walletState.transactions = [transaction];
+
+			testHelper.customRender(<TransactionTable />, context);
+
+			// Act:
+			const transactionType = screen.getByText(expectedTransactionType);
+
+			// Assert:
+			expect(transactionType).toBeInTheDocument();
+		};
+
+		it('renders transaction type as Sent when sender is the same as selected account', () => {
+			assertTransactionType({
+				sender: senderAddress,
+				transactionType: 'Transfer'
+			}, 'Sent');
+		});
+
+		it('renders transaction type as Received when sender is not the same as selected account', () => {
+			assertTransactionType({
+				sender: recipientAddress,
+				transactionType: 'Transfer'
+			}, 'Received');
+		});
+
+		it('renders transaction type as transaction type when not Transfer', () => {
+			assertTransactionType({
+				sender: recipientAddress,
+				transactionType: 'Aggregate Bonded'
+			}, 'Aggregate Bonded');
+		});
+
+		describe('transaction status', () => {
+			const assertTransactionStatus = (transaction, expectedTransactionStatus) => {
+				// Arrange:
+				context.walletState.transactions = [transaction];
+
+				testHelper.customRender(<TransactionTable />, context);
+
+				// Act:
+				const transactionStatus = screen.getByText(expectedTransactionStatus);
+
+				// Assert:
+				expect(transactionStatus).toBeInTheDocument();
+			};
+
+			it('renders transaction as Confirmed when height is not null', () => {
+				assertTransactionStatus({
+					height: 1
+				}, 'Confirmed');
+			});
+
+			it('renders transaction as Pending when height is null', () => {
+				assertTransactionStatus({
+					height: null
+				}, 'Pending');
+			});
+		});
+	});
+
+	describe('third column', () => {
+		it('renders transaction amount when amount is not null', () => {
+			// Arrange:
+			context.walletState.transactions = [{
+				amount: 100
+			}];
+
+			testHelper.customRender(<TransactionTable />, context);
+
+			// Act:
+			const amount = screen.getByText('100 XYM');
+
+			// Assert:
+			expect(amount).toBeInTheDocument();
+		});
+
+		it('does not render transaction amount when amount is null', () => {
+			// Arrange + Act:
+			context.walletState.transactions = [{
+				amount: null
+			}];
+
+			testHelper.customRender(<TransactionTable />, context);
+
+			// Assert:
+			const amount = screen.queryByText('null XYM');
```

Comment:
could ``XYM`` get displayed?

Reply:
Nope, when the amount is null, it displays nothing.
The amount is used in transfer transactions.
Example transfer x amount of XYM.